src/sage/graphs/graph_decompositions/slice_decomposition.pyx

```diff
@@ -0,0 +1,970 @@
+            sage: ('c', 'd') in SD.active_edges('a') # c and d in the same slice
+            False
+            sage: ('a', 'u') in SD.active_edges('a') # u not in x-slice of a
+            False
+
+        The set of active edges of every vertex is a partition of the edges::
+
+            sage: from itertools import chain
+            sage: E = list(chain(*(SD.active_edges(v) for v in G)))
+            sage: G.size() == len(E) == len(set(E)) \
+            ....:   and all(G.has_edge(u, w) for v in G for u, w in SD.active_edges(v))
+            True
+
+        TESTS:
+
+            sage: SD = graphs.PetersenGraph().slice_decomposition()
+            sage: SD.active_edges('Graham')
+            Traceback (most recent call last):
+            ...
+            LookupError: vertex (Graham) does not appear in the slice decomposition
+        """
+        if v not in self.sigma_inv:
+            raise LookupError(f"vertex ({v}) does not appear in the slice "
+                              "decomposition")
+        i = self.sigma_inv[v]
+        return self._xslice_active_edges(i)
+
+    def _slice(self, idx):
+        r"""
+        This method is for internal use only
+        """
+        return list(self.sigma[idx:idx+self.xslice_len[idx]])
+
+    def _xslice_sequence(self, idx):
+        r"""
+        This method is for internal use only
+        """
+        l = self.xslice_len[idx]
+        S = [ [self.sigma[idx]] ]
+        j = idx + 1
+        while j < idx + l:
+            lj = self.xslice_len[j]
+            S.append(list(self.sigma[j:j+lj]))
+            j += lj
+        assert j == idx + l, "slice decomposition is ill-formed"
+        return S
+
+    def _xslice_lex_label(self, idx):
+        r"""
+        This method is for internal use only
+        """
+        return list(self.lex_label[idx])
+
+    def _xslice_active_edges(self, idx):
+        r"""
+        This method is for internal use only
+        """
+        l = self.xslice_len[idx]
+        llv_prefix = len(self.lex_label[idx])
+        A = []
+        j = idx + 1
+        while j < idx + l:
+            lj = self.xslice_len[j]
+            llj = self.lex_label[j]
+            for u in self.sigma[j:j+lj]:
+                for w in llj[llv_prefix:]:
+                    A.append((w, u))
+            j += lj
+        assert j == idx + l, "slice decomposition is ill-formed"
+        return A
+
+    def underlying_graph(self):
+        r"""
+        Return the underlying graph corresponding to the slice decomposition.
+
+        If `G` was the graph given as parameter to compute the slice
+        decomposition, the underlying graph corresponds to ``G.to_simple()``,
+        *i.e.*, it is the input graph without loops and multiple edges.
+
+        EXAMPLES:
+
+        ::
+
+            sage: G = Graph('L~mpn~Nrv{^o~_').relabel('abcdefguvwxyz',inplace=False)
+            sage: SD = G.slice_decomposition(initial_vertex='x')
+            sage: SD.underlying_graph() == G
```

Comment:
Are you sure that you want equality and not isomorphic ?

Reply:
Yes I think equality is enough: the underlying graph is built using the same vertices as the vertices of the input graph of `slice_decomposition` and the edges should be the same (not necessarily stored in the same order but it should not matter).

Note that it is not always true that `G.slice_decomposition().underlying_graph() == G`, because the slice decomposition ignores loops, multiedges and labels of edges (and the information is not retrievable from the `SliceDecomposition` object).

The goal of this function was to be able to reconstruct a graph with the given slice decomposition, not necessarily the input graph. I did not want to store a reference to the input graph in the `SliceDecomposition` object because if the graph is mutable, it could change and the slice decomposition could possibly not correspond to the graph anymore.

I am not sure the `underlying_graph` method is necessary, but it seems a easy way to test that the computation of lexicographic labels and actives edges is correct.

Reply:
I see. Then you may add to the description of the method that it is mostly here test that the computation of lexicographic labels and actives edges is correct. ?

Reply:
done in [c78eb04](https://github.com/sagemath/sage/pull/38299/commits/c78eb04724c17faaee2eb1a85adaaf57a576ac05)